openscm/core/parameterset.py

```diff
@@ -215,17 +216,37 @@ def timeseries(  # pylint: disable=too-many-arguments
         ValueError
             Name not given or invalid region
         """
-        timeseries_type = ParameterType.from_timeseries_type(timeseries_type)
-        interpolation = InterpolationType.from_interpolation_type(interpolation)
-        extrapolation = ExtrapolationType.from_extrapolation_type(extrapolation)
         parameter = self._get_or_create_parameter(
             name, self._get_or_create_region(region)
         )
 
-        parameter.attempt_read(timeseries_type, unit, time_points)
-        return TimeseriesView(
-            parameter, unit, time_points, timeseries_type, interpolation, extrapolation
-        )  # TimeseriesView
+        if parameter.unit is not None:
+            # check conversion can be done
+            UnitConverter(parameter.unit, unit)
+
+        if time_points is None:
+            # fill with junk as can be overwritten later anyway
+            _time_points = np.array(
+                [np.datetime64("{}-01-01".format(y)) for y in range(1990, 1995)]
+            )
+        else:
+            _time_points = time_points
+
+        timeseries_type = ParameterType.from_timeseries_type(timeseries_type)
+        interpolation = InterpolationType.from_interpolation_type(interpolation)
+        extrapolation = ExtrapolationType.from_extrapolation_type(extrapolation)
+        parameter.attempt_read(timeseries_type, unit, _time_points)
+        if time_points is not None:
+            return TimeseriesView(
+                parameter,
+                unit,
+                time_points,
+                timeseries_type,
+                interpolation,
+                extrapolation,
+            )  # TimeseriesView
+
+        return ParameterInfo(parameter)  # no time points so can't get values
```

Comment:
This should actually *always* return a `TimeseriesView`. That should then handle potentially `None` `time_points`.

Reply:
Ok cool, think that solves above two comments too then.